### DISCUSSION TITLE
AWS deployment details ?

### DISCUSSION THREAD

**Discussion by fpavogt** [wmo-im/wis2box#675] [Q&A] [answered]
Dear wis2box gurus,

Not sure if this is the right place to ask, but I'm looking for more info on the AWS deployments options for the wis2box. Is there anyway to get more details on the demonstration deployments (e.g. for Malawi) ? 

What services are being used, specifically ? Is Fargate an option at all (not having the need actively maintain a server would be great) ? 

Are there any tools that would enable such deployements to be easily replicated for other countries (e.g. by means of AWS CLI-compatible sets of instructions of scripts) ?

Just keen to figure out a bit better how things work in practice (inc. S3 buckets, etc ...).

**Comment by maaikelimper** [comment]
Hi Frederic,

The deployments on AWS for countries such as Malawi, involved wis2box running the default docker-compose-stack on an EC2-instance hosted in the Amazon cloud, so these implementations were not using serverless technology.

It would be great if you could help us contribute to the wis2box-community by providing AWS-CLI-compatible sets of instructions of scripts and I can provide assistance by explaining the different wis2box-services in the default docker-compose stack.

The wis2box-storage is using MinIO which can be replaced by S3, while the other services could, in theory, be deployed as containers on Fargate. Unfortunately I have no experience running the wis2box-services in Fargate but since all services are containerized it should be possible to run these in a serverless environment. 

So in short, no sorry I have no scripts or instructions to get started but I am happy to help you get started with this.

@tomkralidis : I'm reminded that I had promised David Podeur for similar help to run wis2box on Kubernetes. Do you have any suggestion on how to assist users with serverless implementation of wis2box ?

kind regards,
Maaike

**Reply by fpavogt** [reply]
Hi @maaikelimper, thanks for the feedback ! 

There's a growing interest in toying around "wis2box on AWS" in my team, as part of the various capacity development activities we are (and expect to be) involved in. 

I still need to go and check/ask for the proper resources to conduct these tests - your reply above is perfect to help me shape a proper request. Once granted, we'd certainly be keen to benefit from your expertise to get started, and also happy to feed whatever we learn/develop back to the community.

**Reply by fpavogt** [reply]
One more question @maaikelimper,

How much effort would you say is needed to deploy a "Malawi-like" solution from scratch, for someone entirely new to wis2box, but expert in Python and some knowledge of AWS ? Even a rough guesstimate would be useful to ask for enough human-resources on my end.

**Reply by jmaxmarno** [reply]
Hi Frederic,
  We have a wis2box running on an AWS EC2 instance. Additionally we have an implementation of a WIS2 global cache that is running on AWS managed services including an ECS Fargate cluster for the mqtt client and mqtt broker components. I'm happy to answer any other questions, feel free to send me an email (max.marno@synopticdata.com)

**Reply by fpavogt** [reply]
Thanks for the info @jmaxmarno ! I might definitely get in touch once we start toying around the code for real.

**Reply by maaikelimper** [reply]
Hi @jmaxmarno , would you be able to share the scripts you used to the setup the MQTT client and MQTT broker components on ECS Fargate here on GitHub ?

**Reply by maaikelimper** [reply, marked as answer]
@fpavogt to answer your question:
"How much effort would you say is needed to deploy a "Malawi-like" solution from scratch"

For someone that is comfortable with Linux and EC2 this is should be very little effort (1 day of work?) 

Here is how to quickly deploy and configure the wis2box as was done for Malawi/Zambia/Namibia:

Fire-up an EC2-instance (t3a.medium or greater) and follow the instructions in the user guide :
https://docs.wis2box.wis.wmo.int/en/latest/user/getting-started.html
https://docs.wis2box.wis.wmo.int/en/latest/user/setup.html

Once the wis2box is deployed and configured you need to find a way to send data into the wis2box-incoming bucket, the solution for which might be different depending on your input-source. You could use a python-script that copies data from the input-source and sends it to the wis2box. An example how to upload data to the wis2box-incoming bucket using python is described here:
https://docs.wis2box.wis.wmo.int/en/latest/user/data-ingest.html#automating-data-ingestion

**Comment by ohmer** [comment]
Hi there,

Has anyone considered an implementation on AWS ECS? I'd love to run wis2box on a serverless platform and Kubernetes seems overkill for the use case. I might work on an ECS/Fargate implementation soon myself, just trying to avoid reinventing the wheel ;-)

**Reply by fpavogt** [reply]
@ohmer: on my end, I can confirm that I have not yet had the time to work on this.

**Reply by maaikelimper** [reply]
From my side, no, I have not yet heard about any serverless implementation of wis2box. 

Only serverless-example I can think is that I helped Singapore to make a Lambda-compatible version of synop2bufr:
https://github.com/World-Meteorological-Organization/synop2bufr/tree/main/aws-lambda

@ohmer @fpavogt while I don't have the resources to work directly on developing wis2box for a serverless platform, I would gladly provide assistance, so please contact me if you need help ...
